pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -178,13 +179,57 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	err = c.Watch(
 		&source.Kind{Type: &corev1.Secret{}},
 		&handler.EnqueueRequestsFromMapFunc{
-			ToRequests: mapFn,
+			ToRequests: targetCredSecretMapFunc,
 		},
 		p)
 	if err != nil {
 		return err
 	}
 
+	// allCredRequestsMapFn simply looks up all CredentialsRequests and requests they be reconciled.
+	allCredRequestsMapFn := handler.ToRequestsFunc(
+		func(a handler.MapObject) []reconcile.Request {
+			log.Info("requeueing all CredentialsRequests")
+			crs := &minterv1.CredentialsRequestList{}
+			err := mgr.GetClient().List(context.TODO(), crs)
+			var requests []reconcile.Request
+			if err != nil {
+				log.WithError(err).Error("error listing all cred requests for requeue")
+				return requests
+			}
+			for _, cr := range crs.Items {
+				requests = append(requests, reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Name:      cr.Name,
+						Namespace: cr.Namespace,
+					},
+				})
+			}
+			return requests
+		})
+
+	adminCredSecretPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isAdminCredSecret(e.MetaNew.GetNamespace(), e.MetaNew.GetName())
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isAdminCredSecret(e.Meta.GetNamespace(), e.Meta.GetName())
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isAdminCredSecret(e.Meta.GetNamespace(), e.Meta.GetName())
+		},
+	}
+	// Watch Secrets and reconcile if we see an event for an admin credential secret in kube-system.
+	err = c.Watch(
+		&source.Kind{Type: &corev1.Secret{}},
+		&handler.EnqueueRequestsFromMapFunc{
+			ToRequests: allCredRequestsMapFn,
+		},
+		adminCredSecretPredicate)
```

Comment:
Not sure about this but is there a need to reconcile all the credentials requests here? If suppose only AWS admin cred has been updated why do we need to reconcile GCP cred requests?

Reply:
It's cleaner IMO to just reconcile everything, the ones that do not match your cloud provider are currently a no-op, so there's virtually no cost to doing so. In the future we may have scenarios where CCO is managing creds for all clusters in Hive from the Hive cluster, and in this case we'd want to reconcile across clouds.